src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -115,84 +123,96 @@ async function onLocalInvokeCommand({
+            'Preparing to run {0} locally...',
+            lambdaLocalInvokeParams.handlerName
+        )
 
-            const relativeFunctionHandler = getRelativeFunctionHandler({
-                handlerName,
-                runtime,
-                handlerFileRelativePath
-            })
+        const baseBuildDir = await makeBuildDir()
+        const codeDir = path.dirname(lambdaLocalInvokeParams.document.uri.fsPath)
+        const documentUri = lambdaLocalInvokeParams.document.uri
+        const handlerName = lambdaLocalInvokeParams.handlerName
 
-            const lambdaInfo = await getLambdaInfoFromExistingTemplate({
-                workspaceUri: lambdaLocalInvokeParams.workspaceFolder.uri,
-                relativeOriginalFunctionHandler: relativeFunctionHandler
-            })
+        const inputTemplatePath = await makeInputTemplate({
+            baseBuildDir,
+            codeDir,
+            relativeFunctionHandler: handlerName,
+            runtime,
+        })
 
-            const properties = lambdaInfo ? lambdaInfo.resource.Properties : undefined
-            const codeDir = properties ? path.join(templateDir, properties.CodeUri) : templateDir
+        const buildArgs: ExecuteSamBuildArguments = {
+            baseBuildDir,
+            channelLogger,
+            codeDir,
+            inputTemplatePath,
+            samProcessInvoker: processInvoker,
+        }
+        if (lambdaLocalInvokeParams.isDebug) {
+            buildArgs.environmentVariables = {
+                SAM_BUILD_MODE: 'debug'
+            }
+        }
+        const samTemplatePath: string = await executeSamBuild(buildArgs)
+
+        const invokeArgs: InvokeLambdaFunctionArguments = {
+            baseBuildDir,
+            documentUri,
+            originalHandlerName: handlerName,
+            handlerName,
+            originalSamTemplatePath: inputTemplatePath,
+            samTemplatePath,
+            runtime,
+        }
 
-            const inputTemplatePath = await makeInputTemplate({
-                baseBuildDir,
-                codeDir,
-                relativeFunctionHandler,
-                properties,
-                runtime
-            })
+        const invokeContext: InvokeLambdaFunctionContext = {
+            channelLogger,
+            configuration,
+            taskInvoker,
+            telemetryService
+        }
 
-            const samTemplatePath: string = await executeSamBuild({
-                baseBuildDir,
-                channelLogger,
-                codeDir,
-                inputTemplatePath,
-                samProcessInvoker: processInvoker,
+        if (!lambdaLocalInvokeParams.isDebug) {
+            await invokeLambdaFunction(
+                invokeArgs,
+                invokeContext
+            )
+        } else {
+            const codeUri = path.join(
+                path.dirname(lambdaLocalInvokeParams.samTemplate.fsPath),
+                CloudFormation.getCodeUri(resource)
```

Comment:
Is it possible for the resource codeUri to contain an absolute path?

Reply:
✔️